webrender/src/resource_cache.rs

```diff
@@ -935,6 +936,24 @@ impl ResourceCache {
                 let (actual_width, actual_height) =
                     compute_tile_size(image_descriptor, tile_size, tile);
 
+                if let Some(ref mut dirty) = dirty_rect {
+                    *dirty = match dirty.intersection(&DeviceUintRect::new(
+                            DeviceUintPoint::new(
+                                tile.x as u32 * tile_size as u32,
```

Comment:
I suppose this should be `(tile.x * tile_size) as u32`?

Reply:
that would be u16 * u16 which we don't want.

Reply:
ah, I assumed those were floats.